core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -257,10 +273,48 @@ public Snapshot apply() {
           .run(index -> manifestFiles[index] = manifestsWithMetadata.get(manifests.get(index)));
 
       writer.addAll(Arrays.asList(manifestFiles));
+
+      if (manifestListKeyMetadata != null) {
+        writer.close();
+        manifestListSize = writer.length();
+      }
     } catch (IOException e) {
       throw new RuntimeIOException(e, "Failed to write manifest list file");
     }
 
+    ByteBuffer wrappedManifestListKeyMetadata = null;
+    String wrappedKeyEncryptionKey = null;
+    // Wrap manifest list key
+    if (manifestListKeyMetadata != null) {
+      Preconditions.checkState(
+          encryptionManager instanceof StandardEncryptionManager,
+          "Can't get key encryption key for manifest lists - encryption manager %s is not instance of StandardEncryptionManager",
+          encryptionManager.getClass());
+      StandardEncryptionManager standardEncryptionManager =
+          (StandardEncryptionManager) encryptionManager;
```

Comment:
I think that this should only need to call `unwrapKey` if the wrapped key encryption key is not tracked by the encryption manager.

Reply:
I've removed this from SnapshotProducer